src/rt/backtrace/elf.d

```diff
@@ -45,330 +51,39 @@ struct Image
 
     const(ubyte)[] getDebugLineSectionData()
     {
-        auto stringSectionHeader = ElfSectionHeader(&file, file.ehdr.e_shstrndx);
-        auto stringSection = ElfSection(&file, &stringSectionHeader);
-
-        auto dbgSectionIndex = findSectionByName(&file, &stringSection, ".debug_line");
-        if (dbgSectionIndex != -1)
-        {
-            auto dbgSectionHeader = ElfSectionHeader(&file, dbgSectionIndex);
-            // we don't support compressed debug sections
-            if ((dbgSectionHeader.shdr.sh_flags & SHF_COMPRESSED) != 0)
-                return null;
-            // debug_line section found and loaded
-            return ElfSection(&file, &dbgSectionHeader);
-        }
-
-        return null;
+        ElfSectionHeader dbgSectionHeader;
+        if (!file.findSectionHeaderByName(".debug_line", dbgSectionHeader))
+            return null;
+
+        // we don't support compressed debug sections
+        if ((dbgSectionHeader.shdr.sh_flags & SHF_COMPRESSED) != 0)
+            return null;
+
+        auto dbgSection = ElfSection(file, dbgSectionHeader);
+        const sectionData = cast(const(ubyte)[]) dbgSection.get();
+        // do not munmap() the section data to be returned
+        import core.stdc.string;
```

Comment:
I think it's a but unusual to have an import in the middle like this.

Reply:
This is a hack anyway and leaks memory (it already is or at least was when I wrote this; there's a bugzilla issue about it).

This `getDebugLineSectionData()` should probably be converted to a `processDebugLineSectionData(void delegate(const(ubyte)[] data) processor)` or so, so that the method can free the data itself afterwards.